src/transformers/utils/bitsandbytes.py

```diff
@@ -264,29 +264,32 @@ def get_keys_to_not_convert(model):
     else:
         tied_keys = sum(tied_params, [])
     has_tied_params = len(tied_keys) > 0
-
-    # Check if it is a base model
-    is_base_model = not hasattr(model, model.base_model_prefix)
-
-    # Ignore this for base models (BertModel, GPT2Model, etc.)
-    if (not has_tied_params) and is_base_model:
+    if not has_tied_params:
+        # if the model has no tied parameters, we check if the output embeddings are the same as the input embeddings
+        # if they are the same, it just means that the model is using the same weights for the input and output
+        if id(model.get_output_embeddings()) == id(model.get_input_embeddings()):
+            has_tied_params = True
```

Comment:
```suggestion
    # This hack is to support the models that are on the Hub `trust_remote_code=True` that did not properly 
    # implemented the tied weights
    if not has_tied_params:
        if id(model.get_output_embeddings()) == id(model.get_input_embeddings()):
            has_tied_params = True
```

Reply:
This is to support models that use trust remote code and did not properly implemented tied weights right?

Reply:
yes.